routers/web/repo/actions/actions.go

```diff
@@ -267,6 +271,36 @@ func List(ctx *context.Context) {
 	ctx.HTML(http.StatusOK, tplListActions)
 }
 
+// loadIsRefDeleted loads the IsRefDeleted field for each run in the list.
+// TODO: move this function to models/actions/run_list.go but now it will result in a circular import.
+func loadIsRefDeleted(ctx *context.Context, runs actions_model.RunList) error {
+	branchRuns := make(map[string][]*actions_model.ActionRun)
+	branches := make(container.Set[string], len(runs))
+	for _, run := range runs {
+		refName := git.RefName(run.Ref)
+		if refName.IsBranch() {
+			branchName := refName.ShortName()
+			run.IsRefDeleted = true // assume it's deleted then if it's found in the database it's not deleted
+			branchRuns[branchName] = append(branchRuns[branchName], run)
```

Comment:
Is it right? What if a branch starts 2 runs?

Reply:
And since the logic seems fragile, I think it needs tests

Reply:
> Is it right? What if a branch starts 2 runs?

A branch could start multiple runs, feature branches maybe updated multiple times and main branches may merge many commits. All these behaviours will trigger action runs.

Reply:
de6160a